cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -180,33 +182,49 @@ func (mgr *packetManagerRest) listPacketDevices() (*Devices, error) {
 	var jsonStr = []byte(``)
 	packetAuthToken := os.Getenv("PACKET_AUTH_TOKEN")
 	url := mgr.baseURL + "/projects/" + mgr.projectID + "/devices"
-	req, _ := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+
+	req, err := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Packet devices: %w", err)
+	}
+
 	req.Header.Set("X-Auth-Token", packetAuthToken)
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
+
 	resp, err := client.Do(req)
 	if err != nil {
-		panic(err)
-		// klog.Fatalf("Error listing nodes: %v", err)
+		return nil, fmt.Errorf("failed to list Packet devices: %w", err)
 	}
+
 	defer resp.Body.Close()
 
 	klog.Infof("response Status: %s", resp.Status)
 
-	var devices Devices
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		var devices Devices
+		if err := json.Unmarshal([]byte(body), &devices); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+		}
 
-	if "200 OK" == resp.Status {
-		body, _ := ioutil.ReadAll(resp.Body)
-		json.Unmarshal([]byte(body), &devices)
 		return &devices, nil
 	}
 
-	return &devices, fmt.Errorf(resp.Status, resp.Body)
+	return nil, fmt.Errorf("failed to list Packet devices: %w", err)
 }
 
 // nodeGroupSize gets the current size of the nodegroup as reported by packet tags.
 func (mgr *packetManagerRest) nodeGroupSize(nodegroup string) (int, error) {
-	devices, _ := mgr.listPacketDevices()
+	devices, err := mgr.listPacketDevices()
+	if err != nil {
+		return 0, err
```

Comment:
I'd annotate the error with `fmt.Errorf` to give it a context. It makes tracing easier in the future.
```suggestion
		return 0, fmt.Errorf("failed to list devices: %w", err)
```